dlt/extract/incremental/transform.py

```diff
@@ -0,0 +1,251 @@
+        incremental_state: IncrementalColumnState,
+        last_value_func: LastValueFunc[TCursorValue],
+        primary_key: Optional[TTableHintTemplate[TColumnNames]]
+    ) -> Tuple[TDataItem, bool, bool]:
+        is_pandas = pd is not None and isinstance(tbl, pd.DataFrame)
+        if is_pandas:
+            tbl = pa.Table.from_pandas(tbl)
+            
+        start_out_of_range = end_out_of_range = False
+        if not tbl:  # row is None or empty arrow table
+            return tbl, start_out_of_range, end_out_of_range
+
+        last_value = incremental_state['last_value']
+
+        if last_value_func is max:
+            compute = pa.compute.max
+            end_compare = pa.compute.less
+            start_compare = pa.compute.greater_equal
+            last_value_compare = pa.compute.greater_equal
+            new_value_compare = pa.compute.greater
+        elif last_value_func is min:
+            compute = pa.compute.min
+            end_compare = pa.compute.greater
+            start_compare = pa.compute.less_equal
+            last_value_compare = pa.compute.less_equal
+            new_value_compare = pa.compute.less
+        else:
+            raise NotImplementedError("Only min or max last_value_func is supported for arrow tables")
+
+        
+        # TODO: Json path support. For now assume the cursor_path is a column name
+        cursor_path = str(cursor_path)
+        # The new max/min value
+        row_value = compute(tbl[cursor_path]).as_py()
+
+        # If end_value is provided, filter to include table rows that are "less" than end_value
+        if end_value is not None:
+            tbl = tbl.filter(end_compare(tbl[cursor_path], end_value))
+            # Is max row value higher than end value?
+            end_out_of_range = not end_compare(row_value, end_value)
+            if end_out_of_range:
+                if is_pandas:
+                    tbl = tbl.to_pandas()
+                return tbl, start_out_of_range, end_out_of_range
+
+        # Filter out all rows which have cursor value equal to last value
+        # and unique id exists in state
+        tbl = tbl.append_column("_dlt_index", pa.array(range(tbl.num_rows)))
+        if last_value is not None:
+            tbl = tbl.filter(last_value_compare(tbl[cursor_path], last_value))
+            # Exclude rows from the table which have unique hashes already seen before
+
+            # Rows with same cursor as stored last value
+            eq_rows = tbl.filter(pa.compute.equal(tbl[cursor_path], last_value))
+            # compute index, unique hash mapping
+            unique_values = self.unique_values(eq_rows, primary_key, resource_name)
+            unique_values = [(i, uq_val) for i, uq_val in unique_values if uq_val not in incremental_state['unique_hashes']]
+            keep_idx = pa.array(i for i, _ in unique_values)
+            # Filter the table
+            tbl = tbl.filter(pa.compute.is_in(tbl["_dlt_index"], keep_idx))
```

Comment:
1. this will keep only records with unique values and remove all the records that are "newer" than unique right? because keep_idx is only on unique values. maybe you should use `is_not_in`?
2. also adding index could be deferred to the moment we need to remove something below. (and maybe primary key could be used for that if present)